tests/conftest.py

```diff
@@ -322,6 +323,84 @@ def log_ocs_version(cluster):
+    """
+    Create a Service in PagerDuty service. The service represents a cluster instance.
+    The service is deleted at the end of the test run.
+
+    Returns:
+        str: PagerDuty service json
+
+    """
+    if config.ENV_DATA["platform"].lower() == constants.OPENSHIFT_DEDICATED_PLATFORM:
+        pagerduty_api = pagerduty.PagerDutyAPI()
+        payload = pagerduty_api.get_service_dict()
+        service_response = pagerduty_api.create("services", payload=payload)
+        msg = f"Request {service_response.request.url} failed"
+        assert service_response.ok, msg
+        service = service_response.json().get("service")
+
+        def teardown():
+            """
+            Delete the service at the end of test run
+            """
+            service_id = service["id"]
+            log.info(f"Deleting service with id {service_id}")
+            delete_response = pagerduty_api.delete(f"services/{service_id}")
+            msg = f"Deletion of service {service_id} failed"
+            assert delete_response.ok, msg
+
+        request.addfinalizer(teardown)
+        return service
+    else:
+        log.info(
+            "PagerDuty service is not created because "
+            f"{constants.OPENSHIFT_DEDICATED_PLATFORM} "
+            "is not used"
+        )
+        return None
+
+
+@pytest.fixture(scope="session", autouse=True)
+def pagerduty_integration(pagerduty_service):
+    """
+    Create a new Pagerduty integration for service from pagerduty_service
+    fixture if it doesn' exist. Update ocs-converged-pagerduty secret with
+    correct integration key. This is currently applicable only for ODF
+    Managed Service.
+
+    """
+    if config.ENV_DATA["platform"].lower() == constants.OPENSHIFT_DEDICATED_PLATFORM:
+        service_id = pagerduty_service["id"]
+        pagerduty_api = pagerduty.PagerDutyAPI()
+
+        log.info(
+            "Looking if Prometheus integration for pagerduty service with id "
+            f"{service_id} exists"
+        )
+        integration_key = None
+        for integration in pagerduty_service.get("integrations"):
+            if integration["summary"] == "Prometheus":
+                log.info(
+                    "Prometheus integration already exists. "
+                    "Skipping creation of new one."
+                )
+                integration_key = integration["integration_key"]
+                break
+
+        if not integration_key:
+            payload = pagerduty_api.get_integration_dict("Prometheus")
+            integration_response = pagerduty_api.create(
+                f"services/{service_id}/integrations", payload=payload
+            )
+            msg = f"Request {integration_response.request.url} failed"
+            assert integration_response.ok, msg
+            integration = integration_response.json().get("integration")
+            integration_key = integration["integration_key"]
+        pagerduty.set_pagerduty_integration_secret(integration_key)
```

Comment:
I would move this code into a function in a pager duty module, and here in the fixture, I would call that function. That said, consider this only if it makes sense.

Reply:
Most of the code is moved into set_pagerduty_integration_secret. In this fixture are mainly checks. I would leave it here.